client/errors.go

```diff
@@ -53,3 +55,28 @@ func newErrMissingResponseField(name string) error {
 func newErrInvalidResponseField(name string, err error) error {
 	return fmt.Errorf("invalid %s field in the response: %w", name, err)
 }
+
+type SignError struct {
+	err error
+}
+
+func NewSignError(err error) SignError {
+	return SignError{err: err}
+}
+
+func (e SignError) Error() string {
+	return fmt.Sprintf("sign err: %v", e.err)
```

Comment:
`s/sign err/signature/` maybe, I prefer to avoid the word "error" in errors, usually it's pretty obvious that we've got an error here. Otherwise (mix some wrapping in) it quickly leads to "abc error: xyz error: asdf failed: that's what really happened` messages.

Reply:
updated to `sign:`